case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -311,4 +323,68 @@ static void validateCaseName(String caseName) {
             throw CaseException.createIllegalCaseName(caseName);
         }
     }
+
+    List<CaseInfos> searchCases(String query) {
+        checkStorageInitialization();
+        List<DateTime> dates = new ArrayList<>();
+        List<EntsoeGeographicalCode> entsoeCodes = new ArrayList<>();
+
+        String decodedQuery;
+        try {
+            decodedQuery = URLDecoder.decode(query, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new PowsyblException("Error when decoding the query string");
+        }
+
+        // the query is an elasticsearch form query, so here it will be :
+        // date:XXX AND tsos:(X)
+        // date:XXX AND tsos:(X OR Y OR Z)
+        //
+        String[] searchFields = decodedQuery.split(" AND ");
+        for (String searchField : searchFields) {
+            String[] field = searchField.split(":");
+            if (field.length > 1) {
+                if (field[0].equals("date")) {
+                    String date = field[1].trim();
+                    if (!StringUtils.isEmpty(date)) {
+                        dates.add(parseDateTime(date));
```

Comment:
Date should be a real data (ISO...) and not a String. I guess when we will switch to Elastic than will allow more complex and specific data search (like based on week day etc)

Reply:
Can you be more explicit ? 
the search is driven by a query string, which contains a part for the date and a part fort the tsos searched.
so the query string is something like : 'q=date:XXXX AND tsos:(FR OR ES OR PT)'

Reply:
So, something like '2020-07-02T10:30:00.000' ?

Reply:
It'a little bit annoying, because I split the query with ' AND ' separating string, to separate the date part and the tsos part, and the I split the date part with ':' to extract the date string value ....

Reply:
Ok let's see that when migrating to Elastic search